File: go-sdk/cortex-mem-go/dto/session.go
package dto

// SessionStartRequest starts or resumes a session.
// POST /api/session/start
//
// Wire format (verified against backend SessionController.java):
//
//	{"session_id":"...", "project_path":"/path", "user_id":"..."}
type SessionStartRequest struct {
	SessionID   string `json:"session_id"`
	ProjectPath string `json:"project_path"` // Wire: "project_path" (NOT "cwd" for session start!)
	UserID      string `json:"user_id,omitempty"`
}

// SessionStartResponse is the response from starting a session.
type SessionStartResponse struct {
	SessionDBID  string `json:"session_db_id"`
	SessionID    string `json:"session_id"`
	Context      string `json:"context,omitempty"`
	PromptNumber int    `json:"prompt_number"`
}

// SessionEndRequest signals session end.
// POST /api/ingest/session-end
//
// Wire format (verified against backend IngestionController.java):
//
//	{"session_id":"...", "cwd":"/path", "last_assistant_message":"..."}
type SessionEndRequest struct {
	SessionID            string `json:"session_id"`
	ProjectPath          string `json:"cwd"` // Wire: "cwd"!
	LastAssistantMessage string `json:"last_assistant_message,omitempty"`
}

// UserPromptRequest records a user prompt.
// POST /api/ingest/user-prompt
//
// Wire format (verified against backend IngestionController.java):
//
//	{"session_id":"...", "prompt_text":"...", "prompt_number":1, "cwd":"/path"}
type UserPromptRequest struct {
	SessionID    string `json:"session_id"`
	PromptText   string `json:"prompt_text"`
	ProjectPath  string `json:"cwd"` // Wire: "cwd"!
	PromptNumber int    `json:"prompt_number,omitempty"`
}

File: js-sdk/cortex-mem-js/src/dto/session.ts
// ============================================================
// Session DTOs
// ============================================================

/**
 * Request to start or resume a session.
 * POST /api/session/start
 *
 * Wire format: {"session_id":"...", "project_path":"/path", "user_id":"..."}
 * Note: uses "project_path" (NOT "cwd") for session start.
 */
export interface SessionStartRequest {
  session_id: string;
  project_path: string;
  user_id?: string;
}

/**
 * Response from starting a session.
 */
export interface SessionStartResponse {
  session_db_id: string;
  session_id: string;
  context?: string;
  prompt_number: number;
}

/**
 * Request to end a session.
 * POST /api/ingest/session-end
 *
 * Wire format: {"session_id":"...", "cwd":"/path", "last_assistant_message":"..."}
 * Note: uses "cwd" (NOT "project_path") for session end.
 */
export interface SessionEndRequest {
  session_id: string;
  cwd: string;
  last_assistant_message?: string;
}

/**
 * Request to record a user prompt.
 * POST /api/ingest/user-prompt
 *
 * Wire format: {"session_id":"...", "prompt_text":"...", "cwd":"/path", "prompt_number":1}
 * Note: uses "cwd" (NOT "project_path") for user prompt.
 */
export interface UserPromptRequest {
  session_id: string;
  prompt_text: string;
  cwd: string;
  prompt_number?: number;
}

/**
 * Response from updating session userId.
 * PATCH /api/session/{sessionId}/user
 */
export interface SessionUserUpdateResponse {
  status: string;
  sessionId: string;
  userId: string;
}
